### DISCUSSION TITLE
LoRa Hardware selection

### DISCUSSION THREAD

**Discussion by EDsteve** [LIFsCode/ELOC-3.0#22] [General]
Hi,

i was looking at LoRa modules for a while and couldn't really decide which ones to buy. So i decided to buy a bunch of different ones and hope that @LIFsCode @OOHehir are not disappointed in my selection :)
The main goal is a mesh network. But in the beginning a simple broadcast to a Gateway will do.
I found RAKwireless the much more professional company but also ordered a few eByte modules. Just for comparing it. eByte has a 30dB module which might come in handy.

Here is the list of modules i ordered and should arrive within the next weeks. All of them use the SX1262 chip:
[RAK3172 module](https://store.rakwireless.com/products/wisduo-lpwan-module-rak3172?variant=40014759297222)
[RAK3172 Breakout Board](https://store.rakwireless.com/products/wisduo-breakout-board-rak3272s) 
[eByte E22-900T22S](https://www.aliexpress.com/item/1005001802601881.html?spm=a2g0o.order_list.order_list_main.23.6b0c18024eqwqQ)
[AI-Thinker Ra-01SH](https://www.aliexpress.com/item/1005002561194884.html?spm=a2g0o.order_list.order_list_main.5.6b0c18024eqwqQ)

I ordered two each. Please let me know which ones you like to have for testing. If you think a different module would be better. Let me know and i will order it.

I am also looking into Gateways right now and have the feeling to go with RAKwireless. Before i buy i will comment here again. But of course any experience and suggestions are appreciated.

Cheeers
ED

**Comment by OOHehir** [comment]
@EDsteve

I've no particular preference/ suggestions on LoRa chip, the only thing I believe worth pointing out is that the RAK chips integrate a microprocessor, a [STM Cortex M4](www.st.com/en/microcontrollers-microprocessors/stm32wle5cc.html). This has obviously good & bad points but I think worth highlighting. I have a [dev board](https://wiki.seeedstudio.com/LoRa_E5_Dev_Board/) utilizing a similar processor around but haven't had a chance to use them much.

With regards gateways I installed a [RAK7243](https://www.rakwireless.com/en-us/products/lpwan-gateways-and-concentrators/rak7243) a few years ago, still going. Seems solid enough..

Regards,

**Comment by LIFsCode** [comment]
I would also evaluate a non-mcroprocessor module, i.e. a pure Sx1262 module:
AI thinker RA01SH would be such a module.

Just to be sure that I recall evrything correctly, Plesse check my summary of differences between sx126x and sx127x series of transceivers:

- sx127x: has better availability of modules (and cheaper)
- sx126x: has less power consumption
- I thought sx127x has higher throughput, but not sure about that
- not sure about amplification and rx sensitivity

Feel free to add any points I might missed, just want to make sure we've got all relevant points for the transceiver selection.

I stumbled upon this library which could be a good starting point: https://github.com/nopnop2002/Arduino-LoRa-Ra01S

Fabian

**Comment by EDsteve** [comment]
According to my research i did about these LoRa chips. These are the main differences:

| Feature                | SX127x         | SX126x         |
|------------------------|----------------|----------------|
| Power consumption       | Higher| Lower|
| Transmit Power         | Up to +20 dBm  | Up to +22 dBm  |
| Max Link Budget      | 160dB| 170dB |
| Sensitivity     | -140dB| -148dB|
| Modulation             | LoRa           | LoRa, (G)FSK   |
| Price| Cheaper| More expensive   |
| Form Factor            | Larger         | Smaller        |
| Global Compliance      | Limited        | More extensive |

The main reason for me to choose the SX126x is the transmit power of 2dB more.

I could make a similar table with pros and cons for MCU version VS just the naked LoRa chip. But it seems we don't need/want the extra MCU.
I have ordered the naked [eByte SX1262](https://www.cdebyte.com/products/E22-900MM22S) and the [AI Thinker](https://docs.ai-thinker.com/en/selection_table) which both don't use MCU. That should give you guys something to play with :)
